### PR TITLE
fix: react macro types

### PR DESCRIPTION
**Pull request by vonovak**
# Description

improves macro TS typings

## Types of changes

[//]: # (What types of changes does your code introduce to Lingui?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

Fixes # (issue)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/react/src/Trans.tsx [hunk 1]
```diff
@@ -21,17 +28,14 @@ export type TransProps = {
...
+} & TransRenderCallbackOrComponent
 
 export function Trans(props: TransProps): React.ReactElement<any, any> | null {
   const { i18n, defaultComponent } = useLingui()
   const { render, component, id, message, formats } = props
 
-  const values = { ...(props.values || {}) }
-  const components = { ...(props.components || {}) }
+  const values = { ...props.values }
+  const components = { ...props.components }
```
Comment (thekip) [inline packages/react/src/Trans.tsx:38]: it is because "undefined" could be also spread to the object?

File: packages/react/src/Trans.tsx [hunk 2]
```diff
@@ -21,17 +28,14 @@ export type TransProps = {
...
-}>
+} & TransRenderCallbackOrComponent
 
 export function Trans(props: TransProps): React.ReactElement<any, any> | null {
   const { i18n, defaultComponent } = useLingui()
   const { render, component, id, message, formats } = props
 
-  const values = { ...(props.values || {}) }
-  const components = { ...(props.components || {}) }
+  const values = { ...props.values }
```
Comment (vonovak) [inline packages/react/src/Trans.tsx:37]: spreading both `null` and `undefined` is valid so this works the same

File: packages/macro/index.d.ts [hunk 3]
```diff
@@ -216,12 +216,10 @@ export function defineMessage(
  */
 export const msg: typeof defineMessage
 
-type CommonProps = {
+type CommonProps = TransRenderCallbackOrComponent & {
   id?: string
   comment?: string
   context?: string
-  render?: (props: TransRenderProps) => ReactElement<any, any> | null
-  i18n?: I18n
```
Comment (vonovak) [inline packages/macro/index.d.ts:224]: this was added in https://github.com/lingui/js-lingui/pull/1308

I tested it and (1) doesn't work (2) isn't documented and right now it's easier to remove it

Comment (thekip) [reply packages/macro/index.d.ts:224]: Yes, i also spotted it there, and wrote an issue to clatify this https://github.com/lingui/js-lingui/issues/1433

File: packages/macro/package.json [hunk 4]
```diff
@@ -63,7 +63,7 @@
     "@lingui/message-utils": "4.0.0"
   },
   "peerDependencies": {
-    "@lingui/react": "4.0.0",
+    "@lingui/react": "^4.0.0",
```
Comment (andrii-bodnar) [inline packages/macro/package.json:66]: All Lingui packages use an exact dependency version of other Lingui packages. Do we really need this change?

Comment (vonovak) [reply packages/macro/package.json:66]: okay, I'll revert

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**